src/adapters/appnexus.js

```diff
@@ -46,187 +49,145 @@ var AppNexusAdapter = function AppNexusAdapter() {
-			var timeDiff = endTime - startTime;
-			var trackingPixelFound = false;
-			var trackingUrl = '';
-			for (var curTrackerItem in timeTrackingBuckets) {
-				//find the closest upper bound of defined tracking times
-				if (timeDiff <= curTrackerItem) {
-					trackingPixelFound = true;
-					trackingUrl = timeTrackingBuckets[curTrackerItem];
-					adloader.trackPixel(trackingUrl);
-					break;
-				}
-			}
-			//if we didn't find a bucket, assume use the catch-all time over bucket
-			if (!trackingPixelFound) {
-				trackingUrl = timeTrackerOverMaxBucket;
-				adloader.trackPixel(trackingUrl);
-			}
-
-			utils.logMessage('latency for placement code : ' + placementCode + ' : ' + timeDiff + ' ms.' + ' Tracking URL Fired : ' + trackingUrl);
+		if(!window.apntag){
+			adloader.loadScript(AST_URL,requestAST);
+		}else{
+			requestAST();
 		}
 	}
 
+	function requestAST(){
+		if(!window.apntag){
+      		utils.logError('appnexus', 'ERROR', 'apntag is not present!');
+		}else{
+			utils._each(bids,function(bid){
 
-	function buildJPTCall(bid, callbackId) {
-
-		//determine tag params
-		var placementId = utils.getBidIdParamater('placementId', bid.params);
-		var memberId = utils.getBidIdParamater('memberId', bid.params);
-		var inventoryCode = utils.getBidIdParamater('invCode', bid.params);
-		var query = utils.getBidIdParamater('query', bid.params);
-
-		//build our base tag, based on if we are http or https
-
-		var jptCall = 'http' + ('https:' === document.location.protocol ? 's://secure.adnxs.com/jpt?' : '://ib.adnxs.com/jpt?');
+				//init ast request
+				window.apntag = window.apntag || {};
+    			apntag.anq = apntag.anq || [];
 
-		//var combinedTargetingParamsList = combineTargetingParams(bidOpts);
+    			if(pbjs.logging){
+    				apntag.enableDebug();
+    			}else{
+    				apntag.disableDebug();
+    			}
 
-		//callback is the callback function to call, this should be hard-coded to pbjs.handleCb once AL-107 is released
-		jptCall = utils.tryAppendQueryString(jptCall, 'callback', 'pbjs.handleCB');
-		jptCall = utils.tryAppendQueryString(jptCall, 'callback_uid', callbackId);
+    			apntag.clearRequest();
 
-		//disable PSAs here, as per RAD-503
-		jptCall = utils.tryAppendQueryString(jptCall, 'psa', '0');
-		jptCall = utils.tryAppendQueryString(jptCall, 'id', placementId);
-		jptCall = utils.tryAppendQueryString(jptCall, 'member_id', memberId);
-		jptCall = utils.tryAppendQueryString(jptCall, 'code', inventoryCode);
+    			//build tag
+    			var astTag = buildTag(bid);
+    			bidmanager.pbCallbackMap[astTag.targetId] = bid;
 
-		//sizes takes a bit more logic
-		var sizeQueryString = utils.parseSizesInput(bid.sizes);
-		if (sizeQueryString) {
-			jptCall += sizeQueryString + '&';
-		}
-		//console.log(jptCall);
+    			apntag.anq.push(function(){
 
-		var targetingParams = utils.parseQueryStringParameters(query);
+    		        var requestTag = apntag.defineTag(astTag);
+    		        requestTag.on('adAvailable',function(){
```

Comment:
we should pass an adData object back inside the handler for AST. `requestTag.on('adAvailable',function(adData){` we need to define the object first.